File: benchmarks/fixtures/webapp_swift/Sources/webapp/models/Session.swift
import Foundation

/// Lifecycle state of a session.
enum SessionStatus: Int {
    case active, expired, revoked
}

/// A login session bound to a user.
struct Session {
    let token: String
    let userId: Int
    var status: SessionStatus

    var isActive: Bool {
        return status == .active
    }
}

File: benchmarks/fixtures/webapp_kt/src/main/kotlin/webapp/models/Session.kt
package webapp.models

/** Lifecycle state of a session. */
enum class SessionStatus {
    ACTIVE,
    EXPIRED,
    REVOKED
}

/** A login session bound to a user. */
data class Session(
    val token: String,
    val userId: Int,
    var status: SessionStatus
) {
    val isActive: Boolean
        get() = status == SessionStatus.ACTIVE
}
